cpp/inc/bond/core/apply.h

```diff
@@ -78,13 +78,13 @@ ApplyTransform(const Transform& transform, const T& value)
 
 
 template <typename Transform, typename T>
-bool inline Apply(const Transform& transform, T& value)
+bool Apply(const Transform& transform, T& value)
```

Comment:
The extern template declaration is not allowed for explicit inline functions: 

> Explicit instantiation of a function template or of a member function of a class template cannot use inline or constexpr.

(see _Expicit instantiation_ section in [here](http://en.cppreference.com/w/cpp/language/function_template), because it makes different requirements with regards to internal/external linkage. Note, that this does not mean the function will/may not be inlined in regular cases (in fact, I never use explicit `inline` with template functions because they are already implicitly declared that way, unless it is a full specialization in a header). It simply allows external linkage as well.